### ISSUE TITLE
acts-as-taggable-on: NameError: uninitialized constant ActsAsTaggableOn::TagListType

### ISSUE THREAD

**Issue by deepj** [oracle/truffleruby#1905]
To reproduce

```ruby
# frozen_string_literal: true

require 'bundler/inline'

gemfile(true) do
  source 'https://rubygems.org'

  git_source(:github) { |repo| "https://github.com/#{repo}.git" }

  # Activate the gem you are reporting the issue against.
  gem 'activerecord', '~> 6.0'
  gem 'acts-as-taggable-on'
  gem 'sqlite3'
end

require 'active_record'
ActiveRecord::Base.establish_connection(adapter: "sqlite3", database: ":memory:")

ActiveRecord::Schema.define do
  create_table :posts, force: true do |t|
    t.string :skills
  end
end

class Post < ActiveRecord::Base
  acts_as_taggable
end
```

Error:

```
uninitialized constant ActsAsTaggableOn::TagListType
~/.gem/truffleruby/2.6.5/gems/acts-as-taggable-on-6.5.0/lib/acts_as_taggable_on/taggable/core.rb:2:in `require_relative'
~/.gem/truffleruby/2.6.5/gems/acts-as-taggable-on-6.5.0/lib/acts_as_taggable_on/taggable/core.rb:2:in `<top (required)>'
~/.gem/truffleruby/2.6.5/gems/acts-as-taggable-on-6.5.0/lib/acts_as_taggable_on/taggable.rb:95:in `taggable_on'
~/.gem/truffleruby/2.6.5/gems/acts-as-taggable-on-6.5.0/lib/acts_as_taggable_on/taggable.rb:40:in `acts_as_taggable_on'
~/dev/bugs/truffleruby/chatwoot/app/models/conversation.rb:57:in `Conversation'
~/dev/bugs/truffleruby/chatwoot/app/models/conversation.rb:31:in `<top (required)>'
~/.gem/truffleruby/2.6.5/gems/activesupport-6.0.2.1/lib/active_support/dependencies/interlock.rb:14:in `block in loading'
~/.gem/truffleruby/2.6.5/gems/activesupport-6.0.2.1/lib/active_support/concurrency/share_lock.rb:151:in `exclusive'
~/.gem/truffleruby/2.6.5/gems/activesupport-6.0.2.1/lib/active_support/dependencies/interlock.rb:13:in `loading'
~/dev/bugs/truffleruby/chatwoot/db/migrate/20170406092426_add_display_id.rb:3:in `change'
~/.gem/truffleruby/2.6.5/gems/activerecord-6.0.2.1/lib/active_record/migration.rb:828:in `exec_migration'
~/.gem/truffleruby/2.6.5/gems/activerecord-6.0.2.1/lib/active_record/migration.rb:812:in `block (2 levels) in migrate'
~/.gem/truffleruby/2.6.5/gems/activerecord-6.0.2.1/lib/active_record/migration.rb:811:in `block in migrate'
~/.gem/truffleruby/2.6.5/gems/activerecord-6.0.2.1/lib/active_record/connection_adapters/abstract/connection_pool.rb:473:in `with_connection'
~/.gem/truffleruby/2.6.5/gems/activerecord-6.0.2.1/lib/active_record/migration.rb:810:in `migrate'
~/.gem/truffleruby/2.6.5/gems/activerecord-6.0.2.1/lib/active_record/migration.rb:1001:in `migrate'
~/.gem/truffleruby/2.6.5/gems/activerecord-6.0.2.1/lib/active_record/migration.rb:1311:in `block in execute_migration_in_transaction'
~/.gem/truffleruby/2.6.5/gems/activerecord-6.0.2.1/lib/active_record/migration.rb:1362:in `block in ddl_transaction'
~/.gem/truffleruby/2.6.5/gems/activerecord-6.0.2.1/lib/active_record/connection_adapters/abstract/database_statements.rb:281:in `block in transaction'
~/.gem/truffleruby/2.6.5/gems/activerecord-6.0.2.1/lib/active_record/connection_adapters/abstract/transaction.rb:280:in `block in within_new_transaction'
~/.gem/truffleruby/2.6.5/gems/activerecord-6.0.2.1/lib/active_record/connection_adapters/abstract/transaction.rb:278:in `within_new_transaction'
~/.gem/truffleruby/2.6.5/gems/activerecord-6.0.2.1/lib/active_record/connection_adapters/abstract/database_statements.rb:281:in `transaction'
~/.gem/truffleruby/2.6.5/gems/activerecord-6.0.2.1/lib/active_record/transactions.rb:212:in `transaction'
~/.gem/truffleruby/2.6.5/gems/activerecord-6.0.2.1/lib/active_record/migration.rb:1362:in `ddl_transaction'
~/.gem/truffleruby/2.6.5/gems/activerecord-6.0.2.1/lib/active_record/migration.rb:1310:in `execute_migration_in_transaction'
~/.gem/truffleruby/2.6.5/gems/activerecord-6.0.2.1/lib/active_record/migration.rb:1282:in `block in migrate_without_lock'
~/.gem/truffleruby/2.6.5/gems/activerecord-6.0.2.1/lib/active_record/migration.rb:1281:in `each'
~/.gem/truffleruby/2.6.5/gems/activerecord-6.0.2.1/lib/active_record/migration.rb:1281:in `migrate_without_lock'
~/.gem/truffleruby/2.6.5/gems/activerecord-6.0.2.1/lib/active_record/migration.rb:1229:in `block in migrate'
~/.gem/truffleruby/2.6.5/gems/activerecord-6.0.2.1/lib/active_record/migration.rb:1382:in `with_advisory_lock'
~/.gem/truffleruby/2.6.5/gems/activerecord-6.0.2.1/lib/active_record/migration.rb:1229:in `migrate'
~/.gem/truffleruby/2.6.5/gems/activerecord-6.0.2.1/lib/active_record/migration.rb:1061:in `up'
~/.gem/truffleruby/2.6.5/gems/activerecord-6.0.2.1/lib/active_record/migration.rb:1036:in `migrate'
~/.gem/truffleruby/2.6.5/gems/activerecord-6.0.2.1/lib/active_record/tasks/database_tasks.rb:238:in `migrate'
~/.gem/truffleruby/2.6.5/gems/activerecord-6.0.2.1/lib/active_record/railties/databases.rake:86:in `block (3 levels) in <top (required)>'
~/.gem/truffleruby/2.6.5/gems/activerecord-6.0.2.1/lib/active_record/railties/databases.rake:84:in `each'
~/.gem/truffleruby/2.6.5/gems/activerecord-6.0.2.1/lib/active_record/railties/databases.rake:84:in `block (2 levels) in <top (required)>'
~/.gem/truffleruby/2.6.5/gems/railties-6.0.2.1/lib/rails/commands/rake/rake_command.rb:23:in `block in perform'
~/.gem/truffleruby/2.6.5/gems/railties-6.0.2.1/lib/rails/commands/rake/rake_command.rb:20:in `perform'
~/.gem/truffleruby/2.6.5/gems/railties-6.0.2.1/lib/rails/command.rb:48:in `invoke'
~/.gem/truffleruby/2.6.5/gems/railties-6.0.2.1/lib/rails/commands.rb:18:in `<top (required)>'
bin/rails:4:in `require'
bin/rails:4:in `<main>'
```

**Comment by eregon** [comment]
Could you post the backtrace?
It's quite useful in general to find out where the bug comes from and assess the bug without needing to reproduce first.
If there are parts of the backtrace with private code, feel free to replace those with `...`.

**Comment by deepj** [comment]
@eregon I added the backtrace while running something else. But it would be reproducible by above code but due to another bug #1895 is not shown

**Comment by deepj** [comment]
@eregon Can be this a priority as well? It'd allow me to dig more and to find other "anomalies" in TruffleRuby :)

**Comment by eregon** [comment]
OK, I'll try to take a look when I get some time.

**Comment by bjfish** [comment]
I looked at this briefly the other day, some things I noticed: 
- disabling didyoumean may help uncover an error since an error happens in didyoumean code handling the NameError
- I suspected an `autoload` issue due to `autoload_under` usage:
https://github.com/mbleigh/acts-as-taggable-on/blob/950c01072b7ab6227d74dc0d0b112f0135281fc2/lib/acts-as-taggable-on.rb#L27
https://github.com/rails/rails/blob/b2eb1d1c55a59fee1e6c4cba7030d8ceb524267c/activesupport/lib/active_support/dependencies/autoload.rb#L50

**Comment by deepj** [comment]
@bjfish hello, any news about this issue? :)

**Comment by bjfish** [comment]
@deepj I just tried this at commit https://github.com/oracle/truffleruby/commit/b9669c4c1aac9546b2e0b772aa180688a9d3aad7 and was unable to recreate. Can you please confirm?

**Comment by deepj** [comment]
@bjfish you need to rerun migration. At least in my case just

```
dropdb chatwoot
rails db:setup
```

And I got it again

**Comment by eregon** [comment]
This should be fixed by af4c22ac7ce963dda0cae8d78a455b0ba55fcc49
